main/src/library/InputStream.flix

```diff
@@ -0,0 +1,89 @@
+namespace InputStream {
+    ///
+    /// Wraps the `java.io.InputStream.available()` method that returns the number of available bytes
+    ///
+    /// See the Java documentation for more details
+    ///
+    pub def available(stream: InputStream): Result[Int32, Error] & Impure = try {
+        import java.io.InputStream.available();
+
+        let InputStream(javaStream) = stream;
+
+        Ok(javaStream.available())
+    } catch {
+        case _: ##java.io.IOException => Err(IOException)
+    }
+
+    ///
+    /// Wraps the `java.io.InputStream.read()` method that reads a single byte from the stream.
+    ///
+    /// See the Java documentation for more details
+    ///
+    pub def readByte(stream: InputStream): Result[Int32, Error] & Impure = try {
+        import java.io.InputStream.read();
+
+        let InputStream(javaStream) = stream;
+
+        Ok(javaStream.read())
+    } catch {
+        case _: ##java.io.IOException => Err(IOException)
+    }
+
+    ///
+    /// Wraps the `java.io.InputStream.read(byte[])` method that reads a number of bytes from the stream.
+    ///
+    /// See the Java documentation for more details
+    ///
+    pub def readBuffer(buffer: Array[Int8], stream: InputStream): Result[Int32, Error] & Impure = try {
+        import java.io.InputStream.read(Array[Int8]);
+
+        let InputStream(javaStream) = stream;
+
+        Ok(javaStream.read(buffer))
+    } catch {
+        case _: ##java.io.IOException => Err(IOException),
+        case _: ##java.lang.NullPointerException => Err(NullPointerException)
+    }
+
+    ///
+    /// Wraps the `java.io.InputStream.read(byte[], int, int)` method that reads a specified number of bytes from the stream.
+    ///
+    /// See the Java documentation for more details
+    ///
+    pub def readWithLength(buffer: Array[Int8], offset: Int32, length: Int32, stream: InputStream): Result[Int32, Error] & Impure = try {
+        import java.io.InputStream.read(Array[Int8], Int32, Int32);
+
+        let InputStream(javaStream) = stream;
+        let readBytes = javaStream.read(buffer, offset, length);
+
+        Ok(readBytes)
+    } catch {
+        case _: ##java.io.IOException => Err(IOException)
+        case _: ##java.lang.NullPointerException => Err(NullPointerException)
```

Comment:
How can this happen?

Reply:
It happens when `buffer` is null, which, when thinking about it for a second, can't happen. I'll remove the case.